consensus/polybft/fsm.go

```diff
@@ -88,17 +86,8 @@ type fsm struct {
 	logger hcf.Logger // The logger object
 }
 
-func (f *fsm) Init(info *pbft.RoundInfo) {
-	err := f.blockBuilder.Reset()
-	if err != nil {
-		panic(err) // TODO: handle differently
-	}
-
-	f.commitmentToSaveOnRegister = nil
```

Comment:
I think we still need to reset commitment in fsm on round change.

Reply:
reverted to previous behaviour https://github.com/0xPolygon/polygon-edge/commit/f0d08eb1a44c7b7b182c3f1757dc2e855bee7d57

Reply:
Let's see if there is an easy way to recognize that round changed so that we can reset this `commitment` field on `fsm`. As it is now, it is not reset, since we are not calling any hook or function on round change.